### PR TITLE
Add fedora and update other linux packages dependencies for substrate & the FE template

### PR DESCRIPTION
**Pull request by grenade**
the following command / package mappings from the apt (ubuntu/debian) instructions were used:

* `sudo apt update` -> `sudo dnf update`
* `sudo apt install` -> `sudo dnf install`
  * `cmake` -> `cmake`
  * `pkg-config` -> `pkgconf-pkg-config`
  * `libssl-dev` -> `openssl-devel`
  * `git` -> `git`
  * `build-essential` -> `make automake gcc gcc-c++ kernel-devel` (see: https://unix.stackexchange.com/a/1344/39654)
  * `clang` -> `clang`
  * `libclang-dev` -> `clang-devel`
  * `curl` -> `curl`
  * `libz-dev` -> `zlib-devel`

- [x] Are the audience and objective of the document clear? E.g. a document for developers that should teach them about transaction fees.
- [x] Is the writing:
  - [x] Clear: No jargon.
  - [x] Precise: No ambiguous meanings.
  - [x] Concise: Free of superfluous detail.
- [x] Does it follow our style guide?
- [x] If this is a new page, does the PR include the appropriate infrastructure, e.g. adding the page to a sidebar?
- [x] Build the page ($ cd website && yarn start). Does it render properly? E.g. no funny lists or formatting.
- [x] Do links go to rustdocs or devhub articles rather than code?
- [x] If this PR addresses an issue in the queue, have you referenced it in the description?

### CODE REVIEW

**Review by NukeManDan — APPROVED** [review]
I don't have a Fedora machine to test this on, but LGTM. Although, I do not think that you must have cmake & automake & make - these were deprecated in v2.0.0 that used the `makefile` flow.

File: docs/knowledgebase/getting-started/index.md
Comment (NukeManDan) [inline docs/knowledgebase/getting-started/index.md:52]:
```suggestion
sudo dnf install clang clang-devel curl gcc gcc-c++ git kernel-devel openssl-devel pkgconf-pkg-config zlib-devel
```
Not sure if you really need all of this either TBH... `curl` could be optional for instance. clang and gcc/g++ are both needed? cc @jimmychu0807 who should we ask about base required deps for us beyond what rust needs?

**Review by NukeManDan — CHANGES_REQUESTED** [review]
want to get this double checked before we commit, and I will try a Fedora VM to find minimal packages.

**Comment by NukeManDan** [comment]
I have installed the node and frontend template, built them, and run them successfully with the now committed deps needed for any linux distro. 

I am not sure, but openssl may not be needed, as it is already installed in all distros I have tested, I assume this is needed for `ws` connections to work. Can I get a last check on this from a core dev that these minimal deps are ok?

**Review by NukeManDan — APPROVED** [review]

**Comment by tomaka** [comment]
Git, OpenSSL, and curl aren't needed to compile Substrate.
`pkg-config`, however, probably is.

**Comment by NukeManDan** [comment]
`git` and `curl` aren't needed to compile Substrate, but they are used in our guide to install (curl rust install script, clone the templates with git)

pkg-config seems to be installed by any ubuntu / fedora base OS, so it's redundant to include here AFAIK. 

IIUC `clang` uses LLVM in a more efficient/compact build for rust, so it's preferred?

I didn't need to add `protoc` to make this work on ubuntu or fedora. I can't find this in my `bin`s either in $PATH on ubuntu... So not needed?

**Comment by NukeManDan** [comment]
So from feedback, this minimal set looks OK - I will keep a lookout for other users having issues if this is incorrect. Thanks guys!

**Pull request merged**
